### PR TITLE
Fix the handling of `b(DID)`.

### PR DESCRIPTION
**Pull request by wujingyue**
According to what we [discussed](https://docs.google.com/document/d/1Lrx_lt09O-POJfIvzTrq-OnfREkYVIbk0ihJzw8amu0/edit#heading=h.366wkzawvgie) on Thursday.

### CODE REVIEW

**Comment by wujingyue** [comment]
!build

File: csrc/ops/utils.cpp [hunk 1]
```diff
@@ -323,6 +323,13 @@ IterDomain* newOutputIterDomain(
       continue;
     }
 
+    NVF_ERROR(
```
Comment (naoyam) [inline csrc/ops/utils.cpp:326]: Was this moved due to the change of the interpretation? Or just for clarity?

File: tests/cpp/test_multidevice_sharding.cpp [hunk 2]
```diff
@@ -230,4 +230,55 @@ TEST_F(MultideviceShardingTest, Issue2758) {
       __FILE__);
 }
 
+TEST_F(MultideviceShardingTest, Broadcast) {
```
Comment (naoyam) [inline tests/cpp/test_multidevice_sharding.cpp:233]: Could you add some comments here? For example, what is this test meant to verify? How would this test fail (or not?) if the change of this PR were not done?

File: csrc/ops/utils.cpp (hunk 1 above)
Comment (wujingyue) [reply csrc/ops/utils.cpp:326]: Without this change, pointwise ops like `set` don't propagate the DID on a broadcast dimension from input to output. Although `b(DID)` is semantically the same as without DID, I think it's better to get the input and the output to agree.

File: tests/cpp/test_multidevice_sharding.cpp (hunk 2 above)
Comment (wujingyue) [reply tests/cpp/test_multidevice_sharding.cpp:237]: Done.

Comment (naoyam) [reply tests/cpp/test_multidevice_sharding.cpp:237]: So, before this PR, if this test is executed with, say, 4 GPUs, the extent of `in->getLogicalDomain()[0]->extent()` and `out->getLogicalDomain()[0]->extent()` would be bound to 4? I'm not sure what would happen then, but I understand that isn't what it should be.

Comment (wujingyue) [reply tests/cpp/test_multidevice_sharding.cpp:237]: > would be bound to 4?

... would be attempted to, and then hit something like "Tried to bind to a constant value: 1 as 4". Overall, we haven't tested a multidevice fusion with broadcasted inputs. Therefore, the bug and this PR.

**Review by naoyam — APPROVED** [review]
LGTM

File: csrc/evaluator_common.cpp [hunk 3]
```diff
@@ -348,19 +348,21 @@ void PrecomputedValues::bindTensorMetaData(
...
-    }
-
-    if (id->hasExpandedExtent()) {
-      Val* extent = id->extent();
-      Val* expanded_extent = id->expandedExtent();
-      bindValue(extent->evaluatorIndex(), 1L);
-      bindValue(expanded_extent->evaluatorIndex(), dim_size);
+    const auto dim_size = tensor.size(static_cast<int64_t>(dim));
+    if (id->isBroadcast()) {
+      // DIDs are ignored.
```
Comment (naoyam) [inline csrc/evaluator_common.cpp:353]: Could you leave a comment why it's ignored? Doesn't need to be here.

**Comment by wujingyue** [comment]
!build

**Pull request merged**